controller/setting_controller.go

```diff
@@ -1561,8 +1559,14 @@ func (info *ClusterInfo) collectVolumesInfo() error {
 	snapshotDataIntegrityCountStruct := newStruct()
 	unmapMarkSnapChainRemovedCountStruct := newStruct()
 	for _, volume := range volumesRO {
-		isVolumeV2DataEngineEnabled := isV2DataEngineEnabled && volume.Spec.BackendStoreDriver == longhorn.BackendStoreDriverTypeV2
-		if !isVolumeV2DataEngineEnabled {
+		backendStoreDriver := types.ValueUnknown
+		if volume.Spec.BackendStoreDriver != "" {
+			backendStoreDriver = util.ConvertToCamel(string(volume.Spec.BackendStoreDriver), "-")
```

Comment:
We need to convert the values to camel so the schema will look like `longhornVolumeBackendStoreDriverV1Count` instead of `longhornVolumeBackendStoreDriverv1Count`. The latter will be converted incorrectly to `longhorn_volume_backend_store_driverv1_count` by the upgrade responder.

Reply:
The converted value is v1 or v2 here instead of the label. 

I would review this again.